tools/wake/cli_options.h

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2022 SiFive, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You should have received a copy of LICENSE.Apache2 along with
+ * this software. If not, you may obtain a copy at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <vector>
+
+#include "gopt/gopt-arg.h"
+#include "gopt/gopt.h"
+
+struct CommandLineOptions {
+  bool check;
+  bool verbose;
+  bool debug;
+  bool quiet;
+  bool wait;
+  bool workspace;
+  bool tty;
+  bool fwarning;
+  int profileh;
+  unsigned int input;
+  unsigned int output;
+  bool last_use;
+  bool last_exe;
+  bool lsp;
+  bool failed;
+  bool script;
+  bool metadata;
+  bool version;
+  bool html;
+  bool global;
+  bool help;
+  bool config;
+  bool debugdb;
+  bool parse;
+  bool tcheck;
+  bool dumpssa;
+  bool optim;
+  bool exports;
+  bool timeline;
+  bool clean;
+  bool list_outputs;
+
+  const char *percent_str;
+  const char *jobs_str;
+  const char *memory_str;
+  const char *heapf;
+  const char *profile;
+  const char *init;
+  const char *chdir;
+  const char *in;
+  const char *exec;
+  const char *job;
+  const char *label;
+  char *shebang;
+  const char *tagdag;
+  const char *tag;
+  const char *api;
+  const char *fd1;
+  const char *fd2;
+  const char *fd3;
+  const char *fd4;
+  const char *fd5;
```

Comment:
Can we make these std::strings? Also why is `shebang` mutable?

Reply:
> Can we make these std::strings?
yes, but can we do that as a different PR?

> Also why is shebang mutable?
Just directly copying over from the previous implementation

Reply:
sounds good to me.